File: matriz/exercicio6.py
# Ler o tabuleiro 3x3
tabuleiro = []

print("Digite o estado do tabuleiro (3x3):")
print("Use -1 para sua peça, 0 para casa vazia, 1 para peça do oponente.")

for i in range(3):
    linha = []
    for j in range(3):
        val = int(input(f"Posição [{i}][{j}]: "))
        while val not in [-1, 0, 1]:
            print("Valor inválido! Digite -1, 0 ou 1.")
            val = int(input(f"Posição [{i}][{j}]: "))
        linha.append(val)
    tabuleiro.append(linha)

jogada = None
for i in range(3):
    for j in range(3):
        if tabuleiro[i][j] == 0:
            jogada = (i, j)
            break
    if jogada is not None:
        break

# resultado
if jogada:
    print(f"\nPróxima jogada sugerida: posição linha {jogada[0]}, coluna {jogada[1]}")
else:
    print("\nNão há jogadas possíveis. O tabuleiro está cheio.")

File: matriz/exercicio6.c
#include <stdio.h>

int main() {
    int tabuleiro[3][3];
    int linha, coluna;
    int jogada_linha = -1, jogada_coluna = -1;

    printf("Digite o estado atual do tabuleiro (3x3):\n");
    printf("Use -1 para sua peça, 1 para oponente, 0 para casa vazia.\n");

    for (linha = 0; linha < 3; linha++) {
        for (coluna = 0; coluna < 3; coluna++) {
            printf("Posicao [%d][%d]: ", linha, coluna);
            scanf("%d", &tabuleiro[linha][coluna]);
        }
    }

    for (linha = 0; linha < 3; linha++) {
        for (coluna = 0; coluna < 3; coluna++) {
            if (tabuleiro[linha][coluna] == 0) {
                jogada_linha = linha;
                jogada_coluna = coluna;
                break;
            }
        }
        if (jogada_linha != -1) {
            break;
        }
    }

    if (jogada_linha != -1 && jogada_coluna != -1) {
        printf("\nProxima jogada sugerida na posicao: [%d][%d]\n", jogada_linha, jogada_coluna);
    } else {
        printf("\nTabuleiro cheio ou nenhuma jogada possivel.\n");
    }

    return 0;
}

